server/template.go

```diff
@@ -78,3 +82,87 @@ input:focus::-moz-placeholder{color:transparent}
 input:focus:-moz-placeholder{color:transparent}
 input:focus:-ms-input-placeholder{color:transparent}`
 )
+
+// indexJS is the javascript for the main index page.
+func indexJS(defaultProvider string) string {
+	var b bytes.Buffer
+
+	b.WriteString(`function sInit() {
+    "use strict";
+    var searchForm = document.forms.search;
```

Comment:
I'm not saying I really approve of this behavior. I generally avoid using IDs for this purpose, but since you're not using `querySelector` or something along those lines, I think it would make sense for the minimalist style of JS you're going for here.